UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Gas.cs

```diff
@@ -9,25 +9,45 @@ public struct Gas
 		// Gas constant
 		public const float R = 8.3144598f;
 
-		public static readonly Gas Plasma = new Gas("Plasma", 200, 40f);
-		public static readonly Gas Oxygen = new Gas("Oxygen", 20, 31.9988f);
-		public static readonly Gas Nitrogen = new Gas("Nitrogen", 20, 28.0134f);
-		public static readonly Gas CarbonDioxide = new Gas("Carbon Dioxide", 30, 44.01f);
+		public static readonly Gas Plasma = new Gas("Plasma", 200, 40f, true, 0.4f, "PlasmaAir", -4);
+		public static readonly Gas Oxygen = new Gas("Oxygen", 20, 31.9988f, false, 0.4f, "NONE", -3);
+		public static readonly Gas Nitrogen = new Gas("Nitrogen", 20, 28.0134f, false, 0.4f, "NONE", -3);
+		public static readonly Gas CarbonDioxide = new Gas("Carbon Dioxide", 30, 44.01f, false, 0.4f, "NONE", -3);
+
+		public static readonly Gas NitrousOxide = new Gas("Nitrous Oxide", 40, 44.01f, true, 0.4f, "NO2", -5);
+		public static readonly Gas Hydrogen = new Gas("Hydrogen", 30, 44.01f, false, 0.4f, "NONE", -3);
+		public static readonly Gas WaterVapor = new Gas("Water Vapor", 40, 44.01f, true, 0.4f, "WaterVapour", -9);
+		public static readonly Gas BZ = new Gas("BZ", 20, 44.01f, false, 0.4f, "NONE", -3);
+		public static readonly Gas Miasma = new Gas("Miasma", 20, 44.01f, true, 0.4f, "Miasma", -7);
+		public static readonly Gas Nitryl = new Gas("Nitryl", 20, 44.01f, true, 0.4f, "Nitryl", -6);
+		public static readonly Gas Tritium = new Gas("Tritium", 10, 44.01f, true, 0.4f, "Tritium", -3);
+		public static readonly Gas HyperNoblium = new Gas("Hyper-Noblium", 2000, 44.01f, false, 0.4f, "NONE", -3);
+		public static readonly Gas Stimulum = new Gas("Stimulum", 5, 44.01f, false, 0.4f, "NONE", -3);
+		public static readonly Gas Pluoxium = new Gas("Pluoxium", 80, 44.01f, false, 0.4f, "NONE", -3);
+		public static readonly Gas Freon = new Gas("Freon", 300, 44.01f, true, 0.4f, "Freon", -8);
```

Comment:
Since the atmospheric system has a Gas[] ( array  so 4 vs 15) with number of gases in the game, idk how much this impacts performance

Reply:
There seems to be a spike for the pipenets but that could probably be optimised.

Reply:
The whole setting the overlays should be pretty optimised, i redid the system until it was good. Hopefully.

Reply:
cool, yeah pipe nets could go over an optimisation,

Reply:
what does the atmospheric thread look like?
I think if I remember is averaging about 13ms

Reply:
Yeah seems like its still 13ms, its just the atmos manager which is spiking.